vm-virtio/src/block.rs

```diff
@@ -722,6 +722,14 @@ impl<T: DiskFile> BlockEpollHandler<T> {
         const EPOLL_EVENTS_LEN: usize = 100;
         let mut events = vec![epoll::Event::new(epoll::Events::empty(), 0); EPOLL_EVENTS_LEN];
 
+        // Before jumping into the epoll loop, check if the device is expected
+        // to be in a paused state. This is helpful for the restore code path
+        // as the device thread should not start processing anything before the
+        // device has been resumed.
+        while paused.load(Ordering::SeqCst) {
+            thread::park();
+        }
+
```

Comment:
Are we doing that to prevent the epoll loop from handling non PAUSE_EVENT before pausing?

Reply:
Yes, I was thinking about the `virtio-net` device and the fact that a TAP event might occur before the pause event. It's really a way to ensure we didn't enter the main loop beforehand.